FIX #22768 filter in mo

Humml87 opened the pull request:
Continue if a empty value ("") on Status field.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
css

andreubisquerra committed:
receipt.php Receipt preview

andreubisquerra committed:
admin/receipt.php Receipt preview

eldy committed:
css

eldy committed:
Look and feel v17

frederic34 committed:
add same translation in many entities

eldy committed:
Merge pull request #22730 from frederic34/overwritetrans

add same translation in many entities

eldy committed:
FIX #22732

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Debug v17

hregis committed:
FIX passing null in "$options" parameter is deprecated

eldy committed:
Merge pull request #22727 from andreubisquerra/develop

NEW Receipt preview in TakePOS setup

eldy committed:
Merge pull request #22736 from hregis/fix_option_passing_null_is_deprecated

FIX passing null in "$options" parameter is deprecated

eldy committed:
Fix alias

eldy committed:
Merge pull request #22599 from fboitel/NEW#22594-unique-vat-id

NEW #22594: can chose if VAT ID is unique or not for third parties

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

bomuux committed:
BUG: can't update product_price_by_qty

a typo in update query.

frederic34 committed:
fetch last_main_doc in societe class

eldy committed:
Debug v17

eldy committed:
Debug v17

kkhelifa-opendsi committed:
NEW: Add new hooks for actioncomm

eldy committed:
Remove non used code for module webhook

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into
develop

aspangaro committed:
NEW Accountancy - Added an option during export to export or not the lettering

aspangaro committed:
NEW Accountancy - Add a graphic option to enable lettering function

eldy committed:
Trans

eldy committed:
Debug v17

eldy committed:
forcerounding can accept MU or MT in price function

eldy committed:
Prepare setnewpassword mode

ns-info90 committed:
Unable to refuse leave #22752

update date refuse

aspangaro committed:
NEW Reception - Add a from/to on date field

aspangaro committed:
NEW Reception - Add a from/to on date field

aspangaro committed:
FIX Accountancy - Delete operation - Problem on SELECT

Commit:
Continue if searchvalue also a "".

aspangaro committed:
FIX Accountancy - the pages of generations of entries are not journals, this term must be removed

aspangaro committed:
NEW Accountancy - Add a way to clean some words when you generate thirdparty accounting account

aspangaro committed:
Merge remote-tracking branch 'upstream/develop' into 17b16

aspangaro committed:
Merge remote-tracking branch 'upstream/develop' into 17b15

aspangaro committed:
Merge remote-tracking branch 'upstream/develop' into 17b14

eldy committed:
Debug v17

eldy committed:
Merge branch '16.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Better visibility on selected basket

eldy committed:
CSS

eldy committed:
Trans

eldy committed:
NEW scheduled job to send unpaid invoice reminder can use the cc and bcc
of template.

eldy committed:
Can force recipient of unpaid invoice email reminder

eldy committed:
Fix filter on entity

eldy committed:
Fix dark theme mode

Commit:
Fix sql error on stripe sepa

UlusoftConsultancy committed:
Add formObjectOptions hook to variants/card.php

Commit:
refresh of the object

eldy committed:
Merge pull request #22779 from Hystepik/develop#2

Fix : Sql error on stripe sepa

eldy committed:
Merge pull request #22772 from aspangaro/17b16

NEW Accountancy - Model Digitaria - Add a way to clean some words when you generate thirdparty accounting account FPC22

eldy committed:
Merge pull request #22780 from UlusoftConsultancy/variants/formObjectOptions

Add formObjectOptions hook to variants/card.php

eldy commented on htdocs/mrp/mo_list.php:
```diff
@@ -244,7 +244,7 @@
 foreach ($search as $key => $val) {
 	if (array_key_exists($key, $object->fields)) {
-		if ($key == 'status' && $search[$key] == -1) {
+		if ($key == 'status' && ($search[$key] == -1) || ($search[$key] == "")) {
```
The condition is mixing && and || in same level. I think you must add ( ) around the 2 test on $search

Humml87 replied:
u right. My feeling for php must grow....

eldy committed:
Merge pull request #22767 from aspangaro/17b14

FIX Accountancy - Delete operation - Problem on SELECT

eldy committed:
Merge pull request #22764 from aspangaro/17b13

NEW Reception - List - Add a from/to on date field

eldy committed:
Merge pull request #22763 from ns-info90/develop

Unable to refuse leave #22752

eldy committed:
Merge pull request #22760 from aspangaro/17b11

NEW Accountancy - Added an option during export to export or not the lettering FPC21

aspangaro committed:
Fix separator

eldy committed:
Merge pull request #22758 from OPEN-DSI/develop_new_actioncomm_hooks

NEW: Add new hooks for actioncomm

eldy committed:
Merge pull request #22739 from frederic34/patch-7

fetch last_main_doc in societe class

eldy committed:
Merge pull request #22737 from bomuux/patch-1

BUG: can't update product_price_by_qty

eldy committed:
Merge pull request #22771 from aspangaro/17b15

FIX Accountancy - the pages of generations of entries are not journals, this term must be removed

eldy committed:
Merge branch 'develop' of github.com:Dolibarr/dolibarr into develop

eldy committed:
Fix better separation label journal and label transfer

eldy committed:
Debug v17

aspangaro committed:
NEW VAT - Admin - Add information on deadline day for submission of VAT declaration

aspangaro committed:
FIX Accountancy - Journal - HTML in title

aspangaro committed:
NEW Invoice - Add french mention on pdf when vat debit option is on

aspangaro committed:
Remove french translation for transifex

Commit:
massaction to affect users on projects

Commit:
quote issue

lamrani002 committed:
Fixe issue in  htdocs/core/boxes/box_services_contracts.php

lamrani002 committed:
fix issue in conta/social/payment

noec764 committed:
FIX: Add Missing "LibStatut" hook in propal, commande, facture

stickler-ci committed:
Fixing style errors.

Commit:
Fix : php 8.1 warnings in ticket

lamrani002 committed:
fix issue for variable does not seem to exist

eldy committed:
Use the new sql syntax to count nb of record for product

eldy committed:
Can show some tab entries as disabled (greyed link)

Commit:
Fix(Scrutinizer) replace return get with the good

lamrani002 committed:
Fix issue variables whithread and errors

eldy committed:
Add tooltip to explain feature to send ticket email

frederic34 committed:
delete extrafields when deleting ecmfiles

frederic34 committed:
clean code

frederic34 committed:
clean code

eldy committed:
Merge pull request #22800 from frederic34/ecmfilesextrafieldsdelete

delete extrafields when deleting ecmfiles

eldy committed:
Merge pull request #22793 from Hystepik/develop#2

Fix : php 8.1 warnings in ticket

eldy committed:
Merge pull request #22792 from noec764/14_FIX_Add_Missing_LibStatut_Hooks

FIX: Add Missing "LibStatut" hooks in propal, commande, facture

Commit:
New : expand/collapse user permission module

eldy committed:
Merge pull request #22789 from fboitel/NEW#735-massaction-user-project

NEW#22735 Massaction to affect users on projects

eldy committed:
Merge pull request #22785 from aspangaro/17b13

NEW Invoice - Add french mention on pdf when vat debit option is on

eldy committed:
Merge pull request #22784 from aspangaro/17b14

FIX Accountancy - Journal - HTML in title

eldy committed:
Merge pull request #22783 from aspangaro/17b16

NEW VAT - Admin - Add information on deadline day for submission of VAT declaration

lamrani002 committed:
fix issue after PR

lamrani002 committed:
fix issue after PR

eldy committed:
Merge pull request #22761 from aspangaro/17b12

NEW Accountancy - Add a graphic option to enable lettering function - FPC21

lamrani002 committed:
fix type of variable  from string to object

lamrani002 committed:
Fix issue scrutinizer property default does not exist in /core/lib/functionsnumtoword.lib.php

lamrani002 committed:
Fix issue scrutinizer in htdocs/core/lib/project.lib.php

aspangaro committed:
FIX Accountancy - When origin invoice is deposit, suggest specific account on credit note

aspangaro committed:
Merge remote-tracking branch 'upstream/develop' into 17b11

aspangaro committed:
FIX Accountancy - When origin invoice is deposit, suggest specific account on credit note

aspangaro committed:
FIX Accountancy - When origin invoice is deposit, suggest specific account on credit note

hregis committed:
NEW can sort and preselected best supplier price

lamrani002 committed:
Fix issue scrutinizer in modules_barcode.class.php duction getNom()

lamrani002 committed:
Fix issue scrutinizer change varible modepaimentid by modepaymentid and ->modepayement by

lamrani002 committed:
Fix issue for variable total_vat to total_tva

lamrani002 committed:
Fix issue scrutinizer in doc_generic_project_odt

lamrani002 committed:
fix issue in with addin version attribute module_project

Commit:
Update perms.php

lamrani002 committed:
Fix issue declare variable

stickler-ci committed:
Fixing style errors.

lamrani002 committed:
fix issue modify variable

Commit:
Fix brackets in the if condition

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into
develop

fappels committed:
Fix warning in reception list third-party type

eldy committed:
Merge branch '16.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch 'develop' of github.com:Dolibarr/dolibarr into develop

eldy committed:
NEW Page for mass stock transfer can be used with no source stock

eldy committed:
Close #22829

eldy committed:
Update list.php

eldy committed:
Merge pull request #22825 from fappels/17_fix_receptionlist_warning

FIX: warning in reception list third-party type column

eldy committed:
Responsive

eldy committed:
Removed dead code

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge pull request #22807 from hregis/new_can_sort_and_preselected_best_supplier_price

NEW can sort and preselected best supplier price

eldy committed:
Merge pull request #22802 from lamrani002/fix_issue_extrafields_class

fix issue Scrutinizer on class extrafields

eldy committed:
Merge pull request #22790 from lamrani002/dev-fix

Fix issue scrutinizer

aspangaro committed:
Merge remote-tracking branch 'upstream/develop' into 17b11

aspangaro committed:
Use getDolGlobalString

BB2A-Anthony committed:
FIX - php 8 warning function lib

eldy committed:
Clean code comment

eldy committed:
NEW Add the SMTP header References on ticket email created by email

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
FIX Email from ticket keep the project link into actioncomm

eldy committed:
Move status of module Webhook as experimental

eldy committed:
Debug v17

eldy committed:
NEW Init module bookcal

eldy committed:
Doc for v17

eldy committed:
Clean package generation (exclude backup files)

eldy committed:
Fix phpcs

lamrani002 committed:
fix params function getDefaultLang 1 to 0

lamrani002 committed:
removing function ProjectLinesPerAction()

lamrani002 committed:
removing function ProjectLinesPerAction()

lamrani002 committed:
Fix problem by removing function projectLinesPerAction()

lamrani002 committed:
Fix by remove method projectLinesPerAction()

eldy committed:
Merge pull request #22830 from bb2a/FIX---php-8-warning-function-lib

FIX - php 8 warning function lib

eldy committed:
Merge pull request #22801 from Hystepik/develop#3

New : expand/collapse user permission module

lamrani002 committed:
Fix by remove method projectLinesPerAction()

eldy committed:
Merge pull request #22812 from lamrani002/fix_issue_html_cerfafr.modules.php

Fix issue Scrutinizer in html cerfafr.modules.php

eldy committed:
Merge pull request #22814 from lamrani002/fix_issue_doc_productOdt

Fix issue Scrutinizer

eldy committed:
Merge pull request #22803 from lamrani002/fix_issue_numToWord

Fix issue scrutinizer htdocs/core/lib/functionsnumtoword.lib.php

lamrani002 committed:
fix issue by removing methods

eldy committed:
Merge pull request #22815 from lamrani002/fix_issue_doc_projectOdt

Fix issue Scrutinizer doc_project_odt

eldy committed:
Doc

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix show thirdparty if it exists only

eldy committed:
Merge pull request #22791 from lamrani002/fix-datePaymentSocialcontribution

fix issue in conta/social/payment

eldy committed:
Merge pull request #22810 from lamrani002/fix_issue_modules_barcode.class.php

Fix issue Scrutinizer in modules_barcode.class.php

lamrani002 committed:
fix the conflit

eldy committed:
Debug v17

eldy committed:
Debug v17

Commit:
Fix(Scrutinizer) replace erros -> errors https://scrutinizer-ci.com/g/Dolibarr/dolibarr/issues/develop/files/htdocs/core/modules/modBlockedLog.class.php?selectedLabels%5B0%5D=9&orderField=lastFound&order=desc&honorSelectedPaths=0

eldy committed:
Better message

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

Commit:
fix : adherent public page

lamrani002 committed:
uncomment attribute withemail in the constuct

eldy committed:
Merge pull request #22840 from Hystepik/develop#2

fix : adherent public page

eldy committed:
Debug v17

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge pull request #22820 from lamrani002/fix_scrutinize_supp_proposal

Fix scrutinize supplier_proposal

eldy committed:
Merge pull request #22817 from lamrani002/fix_issue_mod_project_universal

Fix issue Scrutinizer in core/modules/project/mod_project_universal.php

eldy committed:
Merge pull request #22816 from lamrani002/fix_issue_in_modules_project

Fix issue scrutinizer in modules_project

eldy committed:
Merge pull request #22806 from aspangaro/17b11

FIX Accountancy - When origin invoice is deposit, suggest specific account on credit note

eldy committed:
Debug v17

eldy committed:
Clean code and add context information for onlineSign

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Sec: Hide sensitive data in phpinfo

eldy committed:
Doc

eldy committed:
Debug v17

eldy committed:
Merge pull request #22799 from lamrani002/fix_issue_html_formticket_class

Fix issue variables whithread and errors

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix duplicate data c_action_trigger and c_tva

frederic34 committed:
add hook to fetch product in list

frederic34 committed:
add hook to fetch product in list

eldy committed:
Remove default deprecated constant

frederic34 committed:
hook on thirdparty list

FlorentPoinsaut committed:
Fix uninitialized var

[php:error] [pid 9969] [client 172.18.0.2:60266] PHP Fatal error:  Uncaught TypeError: count(): Argument #1 ($value) must be of type Countable|array, null given in /var/www/html/admin/oauth.php:153\nStack trace:\n#0 {main}\n  thrown in /var/www/html/admin/oauth.php on line 153, referer: https://erp.hype-assets.eu/admin/oauthlogintokens.php

Commit:
fix : php 8.1 warnings

Commit:
update

Commit:
Fix(Scrutinizer) add public

eldy committed:
Merge pull request #22854 from FlorentPoinsaut/patch-1

Fix uninitialized var in oauth

eldy committed:
Merge pull request #22855 from Hystepik/develop#3

fix : php 8.1 warnings

frederic34 committed:
Update list.php

frederic34 committed:
Update list.php

ns-info90 committed:
#22859

eldy committed:
Fix perf index

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge pull request #22865 from FliyFly/bugErrorCode

Fix(Scrutinizer) add public

ns-info90 committed:
Merge branch 'Dolibarr:develop' into develop

frederic34 committed:
add new tags in receipt printer

frederic34 committed:
add new tags in receipt printer

Commit:
Fix(Scrutinizer) fix name function https://scrutinizer-ci.com/g/Dolibarr/dolibarr/issues/develop/files/htdocs/expedition/stats/month.php?selectedLabels%5B0%5D=9&selectedSeverities%5B0%5D=10&orderField=lastFound&order=desc&honorSelectedPaths=0

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into
develop

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

ns-info90 committed:
Merge branch 'Dolibarr:develop' into develop

eldy committed:
Fix installmodules.lock must also block module builder

eldy committed:
Add missing index

eldy committed:
Trans

eldy committed:
FIX deletion of a line of time spent

defrance committed:
if categorie not enabled, we can't use type_account

eldy committed:
Merge pull request #22880 from defrance/patch-251

fix : if categorie not enabled, we can't use object

eldy committed:
Merge pull request #22874 from FliyFly/fixStatsMonth

Fix(Scrutinizer)

eldy committed:
Fix missing abstract methods

eldy committed:
Merge pull request #22870 from FliyFly/scrutinizerApi_recruitment

Fix(Scrutinizer)

eldy committed:
Merge pull request #22868 from frederic34/receiptprinter

add new tags in receipt printer

eldy committed:
Merge pull request #22866 from FliyFly/bugBlockedLog

Fix(Scrutinizer)

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge pull request #22860 from ns-info90/develop

FIX#22859

eldy committed:
Merge pull request #22852 from frederic34/loadstatic

add hook to fetch product in list

eldy committed:
FIX #22843

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix timezone date

eldy committed:
FIX for #22882

eldy committed:
Merge pull request #22794 from lamrani002/fix_issue_vat_number

fix issue for variable does not seem to exist

frederic34 committed:
fix MAIN_GENERATE_SUPPLIER_ORDER_WITH_PICTURE

eldy committed:
Merge pull request #22890 from frederic34/patch-7

fix MAIN_GENERATE_SUPPLIER_ORDER_WITH_PICTURE

eldy committed:
Fix missing default bank account on form

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix warning

eldy committed:
css

eldy committed:
Merge branch '16.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Debug v17

eldy committed:
Debug v17

eldy committed:
Debug v17 - Fix missing delete picto on ODT setup

eldy committed:
Fix perms

eldy committed:
Fix download of sample import files with DoliDroid

eldy committed:
Fix download of sample import files with DoliDroid

eldy committed:
CSS

eldy committed:
Clean no more used files

eldy committed:
Debug v17

eldy committed:
Fix error management

eldy committed:
Debug v17

eldy committed:
Debug v17

eldy committed:
Website

eldy committed:
Enhance default WEBSITE_MAIN_SECURITY_FORCECSP

eldy committed:
Enhance default WEBSITE_MAIN_SECURITY_FORCECSP

Humml87 committed:
Merge branch 'Dolibarr:develop' into 16.0_#22768_Filter_in_MO
